sql: enable tenant testing for zone tests

shubhamdhama opened the pull request:
`` sql: enable tenant testing for `zone_test.go` ``

Updating `TestValidSetShowZones` to use
`createTestServerParamsAllowTenants` instead of `createTestServerParams`
required some investigation. It turned out simpler than expected. Zone
settings for meta ranges aren't supported for secondary tenants, so this
change simply exclude them from setup and validating.

Informs: #140446
Epic: CRDB-48357
Release note: None

`` sql: enable tenant testing for `zone_config_test` ``

Since `GetSpanConfigForKey` is only available in system tenant, it is not
used when running under secondary tenants.

shubhamdhama committed:
sql: enable tenant testing for `zone_config_test`

Since `GetSpanConfigForKey` is only available in system tenant, it is not
used when running under secondary tenants.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/141735)
<!-- Reviewable:end -->

shubhamdhama commented on pkg/config/system.go:
```diff
@@ -378,8 +378,6 @@ func DecodeKeyIntoZoneIDAndSuffix(
 		} else {
 			objectID = keys.SystemRangesID
 		}
-	} else if bytes.HasPrefix(key, keys.TenantPrefix) {
-		objectID = keys.TenantsRangesID
```
I'm a bit unsure about this change.  I think this condition might be outdated.  The problem I ran into was that if the codec belongs to a secondary tenant, the function always uses `keys.TenantsRangesID` for the `objectID`.  This messed up the zone config returned by `getZoneConfig` (and took me way too long to figure out!).

I could also fix this by keeping the condition but only using it for `code.ForSystemTenant`.  But I'm wondering if these pseudo tenant zone configs are still relevant.  Is there a risk in removing this for backward compatibility?

Any help confirming this would be appreciated!

shubhamdhama committed:
sql: enable tenant testing for `zone_test.go`

Updating `TestValidSetShowZones` to use
`createTestServerParamsAllowTenants` instead of `createTestServerParams`
required some investigation. It turned out simpler than expected. Zone
settings for meta ranges aren't supported for secondary tenants, so this
change simply exclude them from setup and validating.

Informs: #140446
Epic: CRDB-38970
Release note: None

rafiss reviewed (approved):
lgtm! thank you

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/141735#-:-OLF_0Y40RQprKpo2geM:btxtjyp)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->
